geoh5py/ui_json/__init__.py

```diff
@@ -18,6 +18,67 @@
 # pylint: disable=unused-import
 # flake8: noqa
 
+from __future__ import annotations
+
+from typing import Any
+
 from .input_file import InputFile
 from .utils import monitored_directory_copy
 from .validation import InputValidation
+
+
+class SetDict(dict):

```

Comment:
Can we move this out of the __init__?

Reply:
Actually, when is it required to assert the unique set?

Reply:
I put it in shared/utils.py

Reply:
> Actually, when is it required to assert the unique set?

Everywhere we set validations.  The goal is to not waste time generating multiple enforcers.